src/uart/mod.rs

```diff
@@ -0,0 +1,81 @@
+//! Using the UART
+use crate::error::{NegativeErrorExt, NumericError};
+use embedded_io;
+use embedded_io::ErrorKind;
+use riot_sys::libc::c_void;
+use riot_sys::uart_init;
+use riot_sys::uart_t;
+use riot_sys::uart_write;
+
+#[derive(Debug)]
+pub struct UARTDevice {
+    dev: uart_t,
+}
+
+#[derive(Debug)]
+pub struct Error(NumericError);
+
+impl From<NumericError> for Error {
+    fn from(err: NumericError) -> Self {
+        Self(err)
+    }
+}
+
+impl embedded_io::Error for Error {
+    fn kind(&self) -> ErrorKind {
+        match -self.0.number() as _ {
+            riot_sys::ENODEV => ErrorKind::NotFound,
+            riot_sys::ENOTSUP => ErrorKind::Unsupported,
+            _ => ErrorKind::Other,
+        }
+    }
+}
+
+impl embedded_io::ErrorType for UARTDevice {
+    type Error = Error;
+}
+
+impl UARTDevice {
+    /// Create a new UARTDevice from a RIOT descriptor
+    pub fn new(dev: uart_t) -> Self {
```

Comment:
A uart_t is hard to construct for users who don't also use riot_sys (which has worse stability than riot_wrappers).

IIRC the most up to date constructor style is in https://doc.riot-os.org/rustdoc/latest/riot_wrappers/gpio/struct.GPIO.html: have a `from_c` (mainly for cases when C code gives you the uart_t), and a `from_uart_number(n: u32-or-so)` that uses riot_sys::macro_UART_DEV.

(That that is not documented is a bug on its own; https://github.com/RIOT-OS/rust-documentation-builder/pull/2 fixes that)